overlord/ifacestate/helpers.go

```diff
@@ -361,3 +361,51 @@ func getConns(st *state.State) (map[string]connState, error) {
 func setConns(st *state.State, conns map[string]connState) {
 	st.Set("conns", conns)
 }
+
+// snapsWithSecurityProfiles returns all snaps that have active
+// security profiles: these are either snaps that are active, or about
+// to be active (pending link-snap) with a done setup-profiles
+func snapsWithSecurityProfiles(st *state.State) ([]*snap.Info, error) {
+	infos, err := snapstate.ActiveInfos(st)
+	if err != nil {
+		return nil, err
+	}
+	seen := make(map[string]bool, len(infos))
+	for _, info := range infos {
+		seen[info.Name()] = true
+	}
+	for _, t := range st.Tasks() {
+		if t.Kind() != "link-snap" || t.Status().Ready() {
+			continue
+		}
+		snapsup, err := snapstate.TaskSnapSetup(t)
+		if err != nil {
+			return nil, err
+		}
+		if seen[snapsup.Name()] {
+			continue
+		}
+
+		doneProfiles := false
+		for _, t1 := range t.WaitTasks() {
+			if t1.Kind() == "setup-profiles" && t1.Status() == state.DoneStatus {
+				doneProfiles = true
+				break
+			}
+		}
+		if !doneProfiles {
+			continue
+		}
+
+		snapName := snapsup.Name()
+		seen[snapName] = true
+		snapInfo, err := snap.ReadInfo(snapName, snapsup.SideInfo)
```

Comment:
Do we want to use any of the helpers from snapstate for reading the info file? I think we wanted to cache them at some point, no?

Reply:
this whole package does this using snap.ReadInfo, I'm not sure what helpers we are talking about here